docs/MacOS-DEVELOP.adoc

```diff
@@ -0,0 +1,132 @@
+= rdiff-backup: {page-name}
+:page-name: MacOS Development Guide
+:pp: {plus}{plus}
+:sectnums:
+:toc: macro
+
+link:.[Home,role="button round"] // *{page-name}*
+
+'''''
+
+toc::[]
+
+Some notes for developers and contributors willing to help with rdiff-backup development on macOS, or simply compile rdiff-backup from source code.
+
+== Pre-requisites
+
+Here is the list of components required to develop or compile rdiff-backup on macOS.
+
+=== Install Python
+
+On macOS, you have two main options to install Python:
+
+- Download from the official Python website (supports universal2):  
+  https://www.python.org/downloads/macos/
+- Install via Homebrew: brew install python
+
+Verify that Python was installed correctly:
+
+[source,shell]
+----
+which python3
+python3 --version
+----
+
+=== Install Required Build Tools
+
+You may need the following tools for building native libraries:
+
+[source,shell]
+----
+xcode-select --install
+----
+
+=== Build and Install librsync from Source
+
+1. Clone or download the librsync source from GitHub:  
+   https://github.com/rdiff-backup/librsync
+
+2. Build and install it:
+
+[source,shell]
+----
+wget https://github.com/librsync/librsync/releases/download/v2.3.4/librsync-2.3.4.tar.gz
+tar zxvf librsync-2.3.4.tar.gz
+cd librsync
+cmake .
+make
+sudo make install
+----
+
+This should install `librsync` in `/usr/local/lib` and headers in `/usr/local/include`.
+
+To confirm installation:
+
+[source,shell]
+----
+ls /usr/local/lib | grep rsync
+----
+
+== Build, Test and Package rdiff-backup
+
+Now you can clone and set up the rdiff-backup source tree:
+
+[source,shell]
+----
+git clone https://github.com/rdiff-backup/rdiff-backup
+cd rdiff-backup
+----
+
+=== Create and Activate Virtual Environment
+
+[source,shell]
+----
+python3 -m venv .venv
```

Comment:
Actually, I create my venv under `dist` to avoid having to exclude one more file. Would you mind?

Reply:
I believe it's a defacto standard to create a virtualenv under ".venv". It's very common and well integrated in vscode. In addition, I've already include .venv to gitignore. :smile: